miden/tests/integration/operations/decorators/advice.rs

```diff
@@ -117,3 +117,18 @@ fn advice_inject_u64div_conditional_execution() {
     let test = build_test!(source, &[8, 0, 4, 0, 1, 0]);
     test.expect_stack(&[0, 0, 0, 0, 0, 4, 0, 8]);
 }
+
+#[test]
+fn advice_inject_mem() {
+    let source = "begin
+    # write to memory and copy from memory to advice map
+    mem_storew.2 mem_storew.3 adv.mem.2.2
+    # read from advice map to advice tape
+    adv.keyval
+    # copy from advice tape to stack
+    adv_loadw adv_loadw
```

Comment:
I think the example has a couple of issues:
- `mem_storew.2 mem_storew.3` saves `[5, 6, 7, 8]` to both memory locations 2 and 3. This is because `mem_storew` does not remove stored items from the stack.
- `adv_loadw adv_loadw` overwrites the same top 4 elements twice. This is because, `adv_loadw` overwrites the top of the stack rather than pushing elements onto the stack.

These two happen to cancel each other out - so, the test passes. But we should fix both issues.

An easy way to do it would be clear the stack before the `adv.keyval` instruction by executing 8 `drop` instructions.

Reply:
Thank you. Fixed.